src/main/java/org/elasticsearch/common/jna/Natives.java

```diff
@@ -62,6 +65,42 @@ public static void tryMlockall() {
         }
     }
 
+    public static void tryVirtualLock()
+    {
+        Kernel32Library kernel = Kernel32Library.getInstance();
+        Pointer process = null;
+        try {
+            process = kernel.GetCurrentProcess();
+            // By default, Windows limits the number of pages that can be locked.
+            // Thus, we need to first increase the working set size of the JVM by
+            // the amount of memory we wish to lock (Xmx), plus a small overhead (1MB).
+            SizeT size = new SizeT(JvmInfo.jvmInfo().getMem().getHeapInit().getBytes() + (1024 * 1024));
```

Comment:
Actually, the comment is wrong...using `Xms`/`getHeapInit` here is intentional.  The assumption is that they're the same if `mlockall` is enabled.  I think I chose `Xms` to be on the safe side in case they aren't.